File: screen-switcher-sample/src/main/java/com/jaynewstrom/screenswitchersample/third/ThirdScreenDialogFactory.java
package com.jaynewstrom.screenswitchersample.third;

import android.app.Dialog;
import android.content.Context;

import com.jnewstrom.screenswitcher.dialoghub.ScopedDialogFactory;

final class ThirdScreenDialogFactory extends ScopedDialogFactory<ThirdComponent> {
    ThirdScreenDialogFactory(Context scopedContext) {
        super(scopedContext);
    }

    @Override public Dialog createDialog(Context context, ThirdComponent component) {
        return new ThirdScreenDialog(context, component);
    }
}

File: screen-switcher-sample/src/main/java/com/jaynewstrom/screenswitchersample/third/ThirdScreenDialogFactory.kt
package com.jaynewstrom.screenswitchersample.third

import android.app.AlertDialog
import android.app.Dialog
import android.content.Context

import com.jnewstrom.screenswitcher.dialoghub.ScopedDialogFactory
import javax.inject.Inject
import javax.inject.Named

internal class ThirdScreenDialogFactory(scopedContext: Context) : ScopedDialogFactory<ThirdComponent>(scopedContext) {
    public override fun createDialog(context: Context, component: ThirdComponent): Dialog {
        return ThirdScreenDialog(context, component)
    }
}

internal class ThirdScreenDialog(context: Context, component: ThirdComponent) : AlertDialog(context) {
    @Inject @field:Named("dialogMessage") lateinit var message: String

    init {
        component.inject(this)
        setTitle("Third Dialog")
        setMessage(message)
        setButton(AlertDialog.BUTTON_NEGATIVE, "Done") { _, _ -> }
    }
}
